sdk/master_board_sdk/CMakeLists.txt

```
@@ -150,19 +150,11 @@ install(TARGETS master_board_example_imu_data_collection DESTINATION bin)
 
 # --- UNIT TESTS -----------------------------------------------------------
 if(BUILD_TESTING)
-  find_package(Catch2 3.1 QUIET)
-  if(NOT Catch2_FOUND)
-    include(FetchContent)
-    FetchContent_Declare(
-      Catch2
-      GIT_REPOSITORY https://github.com/catchorg/Catch2.git
-      GIT_TAG v3.4.0)
-    FetchContent_MakeAvailable(Catch2)
-  endif()
+  find_package(Catch2)
+  add_definitions(-DFOUND_CATCH2_MAJOR_VERSION=${Catch2_VERSION_MAJOR})
```

```suggestion
  if(Catch2_FOUND)
    add_definitions(-DFOUND_CATCH2_MAJOR_VERSION=${Catch2_VERSION_MAJOR})
  else()
    include(FetchContent)
    FetchContent_Declare(
      Catch2
      GIT_REPOSITORY https://github.com/catchorg/Catch2.git
      GIT_TAG v3.6.0)
    FetchContent_MakeAvailable(Catch2)
    add_definitions(-DFOUND_CATCH2_MAJOR_VERSION=3)
  endif()
```

I don't see the point in removing the fetch part